platforms/jvm/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/daemon/DaemonGroovyCompiler.java

```diff
@@ -78,29 +97,46 @@ protected DaemonForkOptions toDaemonForkOptions(GroovyJavaJointCompileSpec spec)
         Iterable<File> classpath = Iterables.concat(spec.getGroovyClasspath(), antFiles);
         VisitableURLClassLoader.Spec targetGroovyClasspath = new VisitableURLClassLoader.Spec("worker-loader", DefaultClassPath.of(classpath).getAsURLs());
 
-        Collection<File> languageGroovyFiles = classPathRegistry.getClassPath("GROOVY-COMPILER").getAsFiles();
-        VisitableURLClassLoader.Spec compilerClasspath = new VisitableURLClassLoader.Spec("compiler-loader", DefaultClassPath.of(languageGroovyFiles).getAsURLs());
+        ClassPath languageGroovyClasspath = classPathRegistry.getClassPath("GROOVY-COMPILER");
 
         FilteringClassLoader.Spec gradleAndUserFilter = getMinimalGradleFilter();
 
         for (String sharedPackage : SHARED_PACKAGES) {
             gradleAndUserFilter.allowPackage(sharedPackage);
         }
 
-        HierarchicalClassLoaderStructure classLoaderStructure =
-            new HierarchicalClassLoaderStructure(classLoaderRegistry.getGradleWorkerExtensionSpec())
-                .withChild(getMinimalGradleFilter())
-                .withChild(targetGroovyClasspath)
-                .withChild(gradleAndUserFilter)
-                .withChild(compilerClasspath);
-
         JavaForkOptions javaForkOptions = new BaseForkOptionsConverter(forkOptionsFactory).transform(mergeForkOptions(javaOptions, groovyOptions));
         javaForkOptions.setWorkingDir(daemonWorkingDir);
         javaForkOptions.setExecutable(javaOptions.getExecutable());
         if (jvmVersionDetector.getJavaVersion(javaForkOptions.getExecutable()).isJava9Compatible()) {
             javaForkOptions.jvmArgs(JpmsConfiguration.GROOVY_JPMS_ARGS);
+        } else {
+            // In JDK 8 and below, we need to attach the 'tools.jar' to the classpath.
+            File javaExecutable = new File(javaForkOptions.getExecutable());
+            JavaInfo jvm = Jvm.forHome(javaExecutable.getParentFile().getParentFile());
```

Comment:
💭 This is one of the methods I've found, which reuses the `Jvm` class. That being said, I'm open to ideas if we could omit this 2*parent call; 

The reason why this looks different from the [`DaemonJavaCompiler.java`](https://github.com/gradle/gradle/pull/28281/files#diff-e56e568a70f371e8fa937e65f7cb5673480123d1ce9665b18114aa84c9a81686) is because the `ForkingJavaCompileSpec` has a field for the java home, meanwhile, `JavaForkOptions` do not have a home, only an executable path.